tagstudio/src/qt/widgets/tag_box.py

```diff
@@ -27,6 +27,19 @@
 logger = structlog.get_logger(__name__)
 
 
+class TagCompleter(QCompleter):
+    def __init__(self, parent, lib):
+        super().__init__(parent)
+        self.lib = lib
+        self.update([])
+    
+    def update(self, exclude:typing.Iterable[str]):
+        tags = [tag.name for tag in self.lib.tags]
+        for value in exclude:
+            tags.remove(value)
```

Comment:
for performance reasons please use `set` instead of `list`, eg:
```suggestion
    def update(self, exclude: set[str]):
        tags = {tag.name for tag in self.lib.tags}
        tags -= exclude
```